python/ray/_private/node.py

```diff
@@ -179,31 +179,42 @@ def __init__(
             date_str = datetime.datetime.today().strftime("%Y-%m-%d_%H-%M-%S_%f")
             self._session_name = f"session_{date_str}_{os.getpid()}"
         else:
-            session_name = ray._private.utils.internal_kv_get_with_retry(
-                self.get_gcs_client(),
-                "session_name",
-                ray_constants.KV_NAMESPACE_SESSION,
-                num_retries=NUM_REDIS_GET_RETRIES,
-            )
-            self._session_name = ray._private.utils.decode(session_name)
+            if ray_params.session_name is None:
```

Comment:
It's unclear when these are None when they are not. Isn't it supposed to be always not None? (since it is passed from the raylet)?  If not can you comment it make it clear? (like in the driver & ray start, it is None, and it is None for workers)

Reply:
Sure

Reply:
Added assert